tests/integration/test_pip.py

```diff
@@ -206,6 +206,26 @@ def test_pip_packages(
             ["Hello, world!"],
             id="pip_e2e_wheels",
         ),
+        # The test relies on rpm and thus requires rpms.lock.yaml defined.
+        # The lock file could be generated by https://github.com/konflux-ci/rpm-lockfile-prototype
+        # The necessary repo definition for it could be found in UBI image
+        # (/etc/yum.repos.d/ubi.repo) and extracted to a local directory.
+        pytest.param(
+            utils.TestParameters(
+                branch="pip/e2e_rust_extensions",
+                packages=(({"type": "pip"}, {"type": "rpm"})),
+                flags=["--dev-package-managers"],
+                check_output=True,
+                check_deps_checksums=False,
+                check_vendor_checksums=False,
+                expected_exit_code=0,
+                expected_output="",
+            ),
+            # Invocation will fail if there was a failure to build the dependencies.
+            ["python3", "/app/src/test_package_cachi2/main.py"],
+            [],
+            id="pip_e2e_rust_extensions",
+        ),
```

Comment:
I just noticed the test tooks a whopping 5 minutes to execute. Can be addressed in a follow up, but this is going to get frustrating with time. It'd be worth exploring whether the dependency set can be decreased. If nothing else, we should adopt a new tag "expensive" and run ITs in local dev envs without that tag active.

Reply:
Decreasing dependency set will be rather hard since the only dependency the test package has now is `cryptography`. However the very nature of a hermetic build of such complexity results in a test instance downloading half the Internet on each run: to build a Rust extension for Python we need not just the Python package, but also entire dependency trees of both Python and Rust parts (regular plus build ones) and also rpms to supply toolchains and necessary C libraries. This is an inherent problem for most e2e tests, it just happens to manifest itself here more prominently. There are ways to tackle this and I am going to discuss them and their trade-offs during F2F.

Reply:
I know, it was a long shot if there wasn't just one that wasn't needed :).